[CustomCom] [WIP] V3 Rewrite

QingyuChai opened the pull request:
Hello.

I'll be trying to rewrite this.

QingyuChai committed:
init

QingyuChai commented:
Features:

- It will probably be built similar to tags.py and trigger cog
- I will try to add more argument support
- If possible, maybe embed support

tekulvw commented on cogs/customcom/customcom.py:
```diff
@@ -0,0 +1,196 @@
+    def __init__(self, bot):
+        self.bot = bot
+        self.file_path = "data/customcom/commands.json"
+        self.c_commands = dataIO.load_json(self.file_path)
```
Please use the Config object instead of dataIO in V3

QingyuChai committed:
first changes

QingyuChai commented:
Sorry there hasn't been much activity here. I'll try to work on it more.

QingyuChai committed:
first major rewrite

all commands except cc list should be working, next commit(s) will make
it more like tags.py

QingyuChai committed:
removing testcog

rttncraft commented:
Copied from Cog-Board: Something that will allow for customization on both the creators side and the users side. A combination of Custom Command & Choose; I would like to be able to create a sample list for a command and allow the user to be able to pick a number of results to be given on output.
I got the idea from this site: http://pynewbs.com/5a/a/

QingyuChai committed:
ignore stuff

QingyuChai committed:
Merge remote-tracking branch 'Twentysix26/v3' into v3

tekulvw commented:
Can I make some changes to this? Things like...not deleting alias or general lol?

QingyuChai commented:
Wait a second... Did I actually do that. I'll fix that lol

QingyuChai committed:
unignored general and alias

QingyuChai committed:
Merge remote-tracking branch 'origin/v3' into v3

QingyuChai committed:
Merge remote-tracking branch 'Twentysix26/v3' into v3

# Conflicts:
#	cogs/alias/alias.py

QingyuChai committed:
Delete testcog.py

QingyuChai committed:
conform to config rewrite

QingyuChai committed:
rewriting code, using custom command object

QingyuChai committed:
Found out that general was still deleted... lol

QingyuChai committed:
Revert "Found out that general was still deleted... lol"

This reverts commit 5e156ae2bb01a2d1aeb2d7f2f7e90fd1f7d80a60.

QingyuChai commented:
Hello! Sorry to bother. It seems that aside from putting it into .gitignore, I also accidentally _deleted_ general.py. Is there anyway to undo this? Sorry for so many problems, first time making Pull Requests :p

QingyuChai committed:
changed the CommandObj class and how it works

QingyuChai committed:
I missed a colon lol

tekulvw commented:
What's the status on this?

QingyuChai commented:
Sorry for the lack of activity, still working on it. I'll try updating more frequently.

QingyuChai committed:
Merge branch 'V3/develop' into v3

QingyuChai committed:
a bit of type hinting added

QingyuChai committed:
slowly implementing random choices too choose from as a response to the custom command

QingyuChai committed:
Merge branch 'V3/develop' into v3

QingyuChai committed:
Merge branch 'V3/develop' into v3

QingyuChai committed:
Merge remote-tracking branch 'Cog-Creators/V3/develop' into v3

QingyuChai committed:
adding more customisability for random responses

QingyuChai committed:
Merge remote-tracking branch 'origin/v3' into v3

palmtree5 committed:
Merge branch 'V3/develop' into v3

QingyuChai commented:
I'll be working on this next weekend. School started again so my timetable is a bit hectic.

QingyuChai committed:
Merge branch 'V3/develop' into v3

QingyuChai committed:
made random random

QingyuChai committed:
updated cc list command

QingyuChai committed:
Merge branch 'V3/develop' into v3

palmtree5 committed:
Merge branch 'V3/develop' into v3

palmtree5 committed:
[CustomCom] fix imports

palmtree5 reviewed (changes requested):
Probably don't need the zip file that's in here. Also should move customcom into `redbot/cogs`

palmtree5 commented:
Closing in favor of #1003
